### DISCUSSION TITLE
Expandable Animation Bar using React-select

### DISCUSSION THREAD

**Discussion by Saga4** [JedWatson/react-select#4920] [Q&A]
We have a search bar implemented using AsyncSelect.
We are looking to fit the search bar in Navbar now and looking to have something like Quora.com where they have animated expandable search bar.

Please suggest on how to achieve this and if its possible with current version of AsyncSelect?

**Comment by ebonow** [comment]
On quora, it looks like two things happen when the input is focused. Both of these things can be triggered on/off with `onFocus` and `onBlur`

      
![Screen Shot 2021-11-17 at 4 51 18 PM](https://user-images.githubusercontent.com/1007579/142331697-5be4f825-b5eb-434c-9fd4-02e6c79b8453.png)

![Screen Shot 2021-11-17 at 4 51 27 PM](https://user-images.githubusercontent.com/1007579/142331702-916ed61a-8b21-43ea-afe8-3f91669dd99f.png)


1. It looks like the input is merely flexing to fill the space made available by removing items next to it. You can remove them or flex-shrink to hide the content and if your style of the select container is set to flex: `1 1 auto` it should fill the available space.

2. A dark transparent fixed div appears below the Navbar. This can be conditionally rendered or maybe even added as a pseudo element
